src/main/java/org/ngafid/flights/process/CSVFileProcessor.java

```diff
@@ -46,105 +56,353 @@ private CSVFileProcessor(Connection connection, byte[] bytes, String filename, P
+            }
+
+            for (int i = 0; i < secondLineArray.length; i++) {
+                String processedDataType = CSVFileProcessor.extractContentInsideParentheses(secondLineArray[i].strip());
+                dataTypes.add(processedDataType);  // Add the processed result to dataTypes
+                String processedHeader = thirdLineArray[i].strip();
+                headers.add(processedHeader.isEmpty() ? "<Empty>" : processedHeader);  // Add header or "<Empty>" if it's blank
             }
+        }
+    }
 
-        } catch (IOException | FatalFlightFileException | CsvException e) {
-            e.printStackTrace();
+    /**
+     * Calculates md5Hash from a given flight (flight entries)
+     * @param flight
+     * @return md5Hash
+     * @throws FlightProcessingException
+     */
+    private String calculateMd5Hash(List<String[]> flight) throws FlightProcessingException {
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            StringBuilder flightDataBuilder = new StringBuilder();
+            // Append each cell from the flight data to the builder
+            for (String[] flightRow : flight) {
+                for (String cell : flightRow) {
+                    flightDataBuilder.append(cell);
+                }
+            }
+            // Calculate MD5 hash and convert to hex
+            byte[] flightHash = md.digest(flightDataBuilder.toString().getBytes(StandardCharsets.UTF_8));
+            String result = DatatypeConverter.printHexBinary(flightHash).toLowerCase();
+            System.out.println("Md5 hash calculated: " + result);
+            return result;
+        } catch (NoSuchAlgorithmException e) {
             throw new FlightProcessingException(e);
         }
+    }
+
+    /**
+     * Extracts three header lines from a csv file, removes # from the beginning of the second line (if # is present)
+     * @param bufferedReader
+     * @return List of header lines (3 lines expected)
+     * @throws IOException
+     */
+    private List<String> extractHeaderLines(BufferedReader bufferedReader) throws IOException {
+        String flightInformationLine = bufferedReader.readLine();
+        String secondLine = bufferedReader.readLine();
+        String thirdLine = bufferedReader.readLine();
+
+        if(secondLine.startsWith("#")){
+            LOG.info(logTag + " extractHeaderLines " + " # found in the beginning of the second line, removing." );
+            secondLine = secondLine.substring(1);  // Remove the first character (if it's '#')
+        }
 
-        FlightBuilder builder = new CSVFlightBuilder(meta, doubleTimeSeries, stringTimeSeries);
+        List<String> headerLines = new ArrayList<>();
+        headerLines.add(flightInformationLine);
+        headerLines.add(secondLine);
+        headerLines.add(thirdLine);
 
-        LOG.info("Returning flight builder!");
-        return Stream.of(builder);
+        return headerLines;
```

Comment:
This seems like a regression from them original header processing code. I would suggest you conditionally modify the unit headers for G5 data exclusively

Reply:
I needed to have this method so that I have access to the header files within the scope of the parse method. 
It might also be a cleaner solution that enhances readability and maintainability.

I removed the code below so that the header files remain intact ( if that was your concern)

`  if(secondLine.startsWith("#")){
            LOG.info(logTag + " extractHeaderLines " + " # found in the beginning of the second line, removing." );
            secondLine = secondLine.substring(1);  // Remove the first character (if it's '#')
        }`